src/base/bittorrent/private/bandwidthscheduler.cpp

```diff
@@ -24,72 +25,78 @@
  * modify file(s), you may extend this exception to your version of the file(s),
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
- *
- * Contact : chris@qbittorrent.org
  */
 
-#include <QTime>
+#include "bandwidthscheduler.h"
+
 #include <QDateTime>
+#include <QTime>
 
 #include "base/bittorrent/session.h"
 #include "base/preferences.h"
-#include "bandwidthscheduler.h"
 
-BandwidthScheduler::BandwidthScheduler(QObject *parent)
-    : QTimer(parent)
+BandwidthScheduler::BandwidthScheduler(BitTorrent::Session *session)
+    : QObject(session)
+    , m_session(session)
+    , m_lastAlternative(false)
 {
-    // Single shot, we call start() again manually
-    setSingleShot(true);
-    // Connect Signals/Slots
-    connect(this, SIGNAL(timeout()), this, SLOT(start()));
+    connect(&m_timer, &QTimer::timeout, this, &BandwidthScheduler::onTimeout);
 }
 
 void BandwidthScheduler::start()
 {
+    m_lastAlternative = isTimeForAlternative();
+    m_session->setAltGlobalSpeedLimitEnabled(m_lastAlternative);
+
+    // Timeout regularly to accommodate for external system clock changes
+    // eg from the user or from a timesync utility
+    m_timer.start(1500);
+}
+
+bool BandwidthScheduler::isTimeForAlternative()
+{
     const Preferences* const pref = Preferences::instance();
-    bool alt_bw_enabled = BitTorrent::Session::instance()->isAltGlobalSpeedLimitEnabled();
 
     QTime start = pref->getSchedulerStartTime();
     QTime end = pref->getSchedulerEndTime();
     QTime now = QTime::currentTime();
-    int sched_days = pref->getSchedulerDays();
+    int schedulerDays = pref->getSchedulerDays();
     int day = QDateTime::currentDateTime().toLocalTime().date().dayOfWeek();
-    bool new_mode = false;
-    bool reverse = false;
+    bool alternative = false;
 
     if (start > end) {
-        QTime temp = start;
-        start = end;
-        end = temp;
-        reverse = true;
+        std::swap(start, end);
```

Comment:
There is no `<algorithm>` include, might become broken due to a change in other file.